Iter23

Fairness2 opened the pull request.

Fairness2 committed:
iter20.5 | Добавлены тесты для DBStorage

Добавлены тесты для методов DBStorage: GetGauge и GetCounter. Это обеспечивает корректное поведение и улучшает покрытие кода тестами.

Changed files:
- internal/metrics/DBStoragepart3_test.go
- internal/metrics/DBStorage.go

Fairness2 committed:
iter20.6 | Добавлены новые тесты для логгера и DBStorage

Включены тесты для методов логгера и дополнительный набор тестов для DBStorage, чтобы улучшить тестовое покрытие и убедиться в корректной работе кода.

Changed files:
- internal/logger/logger_test.go
- internal/metrics/DBStoragepart3_test.go

Fairness2 committed:
iter20.7 | Добавлена обработка ошибок для размера рабочего процесса, пустого клиента и URL-адреса сервера в sendpool.

Добавлены новые ошибки ErrorWrongWorkerSize, ErrorEmptyClient и ErrorServerURLIsEmpty. Обновлена сигнатура функций New и NewWithClient для возврата ошибок.

Измененные файлы:
- cmd/agent/sendpool/pool.go
- cmd/agent/sendpool/pool_test.go
- cmd/server/handlers/handlemetric/errors.go
- cmd/agent/sendpool/pool1_test.go

Fairness2 committed:
iter21 | Заготовки

Fairness2 committed:
iter21 | Добавлена функция шифрования и дешифрования.

Внедрена поддержка шифрования и дешифрования данных с использованием RSA. Добавлены новые функции для обработки ключей, а также обновлены существующие методы и мидлвары для работы с зашифрованными данными.

Измененные файлы:
- cmd/agent/sendpool/pool.go
- cmd/agent/config/parse.go
- internal/config/keys.go
- cmd/server main.go
- internal/encrypt/decrypter.go
- cmd/server/config/parse.go
- cmd/server/config/config.go
- deployments/keys/private.pem
- deployments/keys/public.pem

Fairness2 committed:
iter22 | Добавлена поддержка чтения конфигурации из файла как для агента, так и для сервера.

Добавлена возможность считывания конфигурации из файла для компонентов agent и server. Реализованы функции парсинга конфигурационного файла, добавлены соответствующие флаги командной строки и переменные окружения.

Измененные файлы:
- cmd/agent/config/parse.go
- internal/config/keys.go
- cmd/agent/config/env.go
- cmd/server/config/fileconfig.go (новый файл)
- cmd/agent/config/fileconfig.go (новый файл)
- internal/config/types.go (новый файл)
- cmd/server/config/parse.go
- cmd/server/config/config.go

Fairness2 committed:
iter23 | Обновил обработку сигналов и тесты для дешифраторов

Переписал обработку сигналов для корректной остановки сервисов, используя `signal.NotifyContext`. Усовершенствовал логику дешифрации, убрал использование ошибки в функции разделения сообщений и добавил тесты для шифрования и дешифрации.

Changed files:
- cmd/server/main.go
- internal/encrypt/decrypter.go
- cmd/agent/main.go
- internal/encrypt/decrypter_test.go

Fairness2 committed:
iter23.1 | Исправил логику восстановления конфигурации.

Обновил условие проверки восстановления, чтобы корректно применять пользовательские настройки. Теперь конфигурация правильно учитывает значение Restore из файла конфигурации.

Список измененных файлов:
cmd/server/config/parse.go

ekubyshin commented on cmd/agent/main.go:
```diff
@@ -73,22 +75,27 @@ func main() {
 	// Ожидаем сигнала завершения Ctrl+C, чтобы корректно завершить работу
-	stop := make(chan os.Signal, 1)
+	<-ctx.Done()
+	/*stop := make(chan os.Signal, 1)
```
Не рабочий код можно сразу удалять, тк вся история хранится в гите, чтобы не засорять код проекта.

Fairness2 committed:
iter23.2 | Удалил закомментированный код обработки сигналов остановки

Удалил закомментированный код, связанный с обработкой сигналов остановки (SIGINT, SIGTERM) в `server` и `agent`. Это улучшает читаемость и поддерживаемость кода.

Список изменённых файлов:
- cmd/server/main.go
- cmd/agent/main.go
